stdlib/stomp/src/main/java/org/ballerinalang/stdlib/stomp/DefaultStompClient.java

```diff
@@ -0,0 +1,188 @@
+            }
+
+    }
+
+    private class ResponseCallback implements CallableUnitCallback {
+        private String message;
+        private String messageId;
+
+        ResponseCallback(String message, String messageId) {
+            this.message = message;
+            this.messageId = messageId;
+        }
+
+        @Override
+        public void notifySuccess() {
+            log.debug("Success notification");
+        }
+
+        @Override
+        public void notifyFailure(BError error) {
+            ErrorHandlerUtils.printError("error: " + BLangVMErrors.getPrintableStackTrace(error));
+        }
+    }
+
+    @Override
+    public void onReceipt(String receiptId) {
+    }
+
+    @Override
+    public void onError(String message, String description) {
+        Resource errorResource = getErrorResource();
+        try {
+            Executor.submit(errorResource, new ResponseCallback(message, description),
+                        null, null, getErrorSignatureParameters(errorResource, message));
+        } catch (BallerinaConnectorException c) {
+                log.error("Error while executing onError resource", c);
+        }
+    }
+
+    @Override
+    public void onCriticalError(Exception e) {
+        System.out.println("Error: " + e);
+    }
+
+    public void registerService(Service service, String destination) {
+        this.serviceRegistry.put(destination, service);
+    }
+
+    public Map getDestinationValue() {
+        return this.serviceRegistry;
+    }
+
+    public Resource getMessageResource(){
+        Service subscriberService = this.serviceRegistry.get("destination");
+        Resource msgResource = null;
+        int count;
+        for (count = 0; count < subscriberService.getResources().length; count++) {
+            // Accessing each element of array
+            String resourceName = subscriberService.getResources()[count].getName();
+            if (resourceName.equals("onMessage")) {
+                Resource onMessageResource = subscriberService.getResources()[count];
+                if (onMessageResource != null) {
+                    msgResource = onMessageResource;
+                }
+            }
+        }
+        return msgResource;
+    }
+
+    public Resource getErrorResource(){
+        // Get service resources by iterating
+        Service subService = this.serviceRegistry.get("destination");
+        Resource errResource = null;
+        int count;
+        for (count = 0; count < subService.getResources().length; count++) {
+            // Accessing each element of array
+            String resourceName = subService.getResources()[count].getName();
+            if (resourceName.equals("onError")) {
+                Resource onErrorResource = subService.getResources()[count];
+                if (onErrorResource != null) {
+                    errResource = onErrorResource;
+                }
+            }
+        }
+        return errResource;
+    }
```

Comment:
Instead of having two methods that have duplicate code, can we use one method and pass the resource name as a parameter

Reply:
fixed